source/tools/configuration.py

```diff
@@ -79,8 +80,11 @@ def uninitialize(cls):
         :return: None
         :rtype: NoneType
         """
-        node_id = SettingList.get_setting_by_code(code='node_id').value
-        if node_id is not None and cls.dir_exists(Configuration.ASD_NODE_LOCATION.format(node_id)):
+        try:
+            node_id = SettingList.get_setting_by_code(code='node_id').value
```

Comment:
There are a lot more occurrences of `get_setting_by_code`, why not implement identical changes there?

Reply:
Only changed the code paths which were actually checking on None